Update CQL driver version to 4.2.0

marco-scoppetta opened the pull request:
The new Java driver has:

- new Maven coordinates (groupId and artifactId)
- new APIs
- does not have Cluster anymore (wrapped in Session)
- some configs are now mandatory, e.g. LocalDatacenter is now mandatory when specifying custom endpoints
- some configs have been removed

NOTES:
- initializing a table seems to be slower than before

Signed-off-by: Marco Scoppetta <marco@grakn.ai>

-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there an issue associated with this PR? Is it referenced in the commit message?
- [ ] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [ ] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you written and/or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](https://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [ ] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?
- [ ] If this PR is a documentation-only change, have you added a `[skip ci]`
  tag to the first line of your commit message to avoid spending CPU cycles in
  Travis CI when no code, tests, or build configuration are modified?

### Note:
Please ensure that once the PR is submitted, you check Travis CI for build issues and submit an update to your PR as soon as possible.

porunov commented:
@marco-scoppetta Thank you for this upgrade!
Please, include `[full build]` in the commit name so that travis would include all Cassandra tests. I.e. just change the commit name to `Update CQL driver version [full build]`.

Ping me when the PR is ready for the review.

farodin91 commented:
Our plan is to remove the project janusgraph-cassandra and the support for cassandra 2. This is required to make the hadoop completely work for cql.

I will started to work on this part today. So, we could combine our work force.

NOTE: Today, i merge testcontainers-cql which introduced a failure on branch 0.4 and master. I'm try to get the fix ready today.

porunov commented:
@marco-scoppetta I have stopped your travis build. Please, rebase your commit from current master state

marco-scoppetta commented:
@porunov it's already based on current master my latest commit

porunov commented:
@marco-scoppetta My mistake, sorry.
I just see that hadoop-test is failing that is why I stopped others.
Here is a failed test:
https://travis-ci.org/JanusGraph/janusgraph/jobs/572677793

porunov commented:
@marco-scoppetta I see that travis build s failing for `cql` tests but this is most likely due to #1545 (not sure yet). The same `cql` tests are failing in `0.3` branch.
See travis build for those tests here: https://travis-ci.org/porunov/janusgraph/builds/572882937
So, most likely those fails are not introduced by your PR.

porunov commented:
@marco-scoppetta Due to the problem with test containers, I had to revert the latest commit. Please exclude additional commits from your PR

porunov commented:
@marco-scoppetta The 4.2.0 version is released. I think it is better to use that version.

farodin91 commented:
Could you squash all commits?

marco-scoppetta commented:
@farodin91 let's first see if tests pass :)

marco-scoppetta commented:
@farodin91 @porunov Also why do you want us to keep squashing commits when you can just "Squash and Merge" at the end? 
I am just curious

farodin91 commented:
The commits also contain Testcontainers commits which are already merge and this make the review harder.

marco-scoppetta commented:
> The commits also contain Testcontainers commits which are already merge and this make the review harder.

@farodin91 that's my mistake in merging, it's not related to squashing a commit

farodin91 commented:
> @farodin91 that's my mistake in merging, it's not related to squashing a commit

These are other commit than in the current master they should be removed completely because they contain bugs like murrur file.

marco-scoppetta commented:
> > @farodin91 that's my mistake in merging, it's not related to squashing a commit
> 
> These are other commit than in the current master they should be removed completely because they contain bugs like murrur file.

I understand your point of view, I've deleted those files, but still it's not related with squashing the commits

porunov commented:
@marco-scoppetta Could you please remove Jan's commits and squash your comments into a single one? It will be much cleaner and easier to review. Looks like CI fails right now but it isn't clear if it fails because of your changes or because of Jan's commits in your PR.

marco-scoppetta commented:
@porunov you can see the files changes in the Files Changes section, squashing commit won't change that, right?

I have already rebased on master, squashing commits only rewrites history, I will fix the tests and then squash

marco-scoppetta commented:
Tests are failing because `NoClassDefFound com/datastax/driver/core/`, so it's clearly my changes, I don't have much times in these days, I am trying to slowing fix the tests at the end of the day

marco-scoppetta commented on janusgraph-hadoop-parent/pom.xml:
```diff
@@ -205,7 +205,13 @@
             <groupId>com.datastax.cassandra</groupId>
             <artifactId>cassandra-driver-core</artifactId>
-            <version>${cassandra-driver.version}</version>
+            <version>3.7.2</version>
```
This is because this class still depends on the old driver: https://github.com/JanusGraph/janusgraph/blob/master/janusgraph-hadoop-parent/janusgraph-hadoop-core/src/main/java/org/janusgraph/hadoop/formats/cassandra/CqlBridgeRecordReader.java#L26

marco-scoppetta replied:
not sure if we still need that class, if we dont I will just remove it

farodin91 replied:
You can delete CqlBridgeRecordReader and remove all explicit tests for it. Basically. call cassandra3

marco-scoppetta commented on janusgraph-cql/src/main/java/org/janusgraph/diskstorage/cql/CQLConfigOptions.java:
```diff
@@ -165,20 +165,6 @@
             ConfigOption.Type.FIXED,
             64);
 
-    ConfigOption<Integer> LOCAL_CORE_CONNECTIONS_PER_HOST = new ConfigOption<>(
```
yeah as soon as tests pass

farodin91 commented:
@marco-scoppetta Would it be okay to use your code as a start to update the driver?

Also, I saw Grakn.io forked JanusGraph. It would intresting to known, if Grakn would you use JanusGraph instead again, for example, some blocker to known would be good.

marco-scoppetta commented:
Hey @farodin91, sorry I am seeing this only now (I am now using another private Github account for a new company). 
Yes I had forked Janus and Grakn is currently using a lighter version of it (mainly because we did not need many features so we removed a lot of stuff and make the code more simple for us) and I had updated the Cassandra driver to 4.2.0 as you can see [here](https://github.com/graknlabs/grakn/blob/master/graph/diskstorage/cql/CQLStoreManager.java), I think you might find useful the settings in line 242 and 243 in that file (I am not sure if you still have that issue of `storeManager.close()` taking a long time, but you do, that's the solution).
Sure use my code as you wish!
